src/makeCalculator.js

```diff
@@ -4,7 +4,68 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  return {
+    result: 0,
+
+    add(value) {
+      this.result += value;
+
+      return this;
+    },
+
+    subtract(value) {
+      this.result -= value;
+
+      return this;
+    },
+
+    multiply(value) {
+      this.result *= value;
+
+      return this;
+    },
+
+    divide(value) {
+      if (value !== 0) {
+        this.result /= value;
+      } else {
+        throw new Error("Can't divide by 0!");
+      }
+
+      return this;
+    },
+
+    operate(operation, value) {
+      if (value === undefined) {
+        throw new Error('No value given');
+      }
+
+      switch (operation) {
```

Comment:
The `operate` method should not check that the callback provided is one of the calculator's own methods. You can simplify this method by just calling the provided function with the value.